src/main/scala/it/unibo/authsim/library/dsl/policy/builders/PolicyBuilder.scala

```diff
@@ -79,18 +98,20 @@ object PolicyBuilder:
 
     override def build: Policy = new Policy:
 
-      override def name: String = PolicyBuilderImpl.this.name
+      override def name: String = BasicPolicyBuilder.this.name match
+        case Some(name) => name
+        case _ => "Policy"
 
-      override def credentialPolicies: Seq[CredentialPolicy] = PolicyBuilderImpl.this._credentialPolicies
+      override def credentialPolicies: Seq[CredentialPolicy] = BasicPolicyBuilder.this._credentialPolicies
 
-      override def cryptographicAlgorithm: Option[CryptographicAlgorithm] = PolicyBuilderImpl.this._cryptographicAlgorithm
+      override def cryptographicAlgorithm: Option[CryptographicAlgorithm] = BasicPolicyBuilder.this._cryptographicAlgorithm
 
-      override def saltPolicy: Option[SaltPolicy] = PolicyBuilderImpl.this._saltPolicy
+      override def saltPolicy: Option[SaltPolicy] = BasicPolicyBuilder.this._saltPolicy
 
-      override def transmissionProtocol: Option[Protocol] = PolicyBuilderImpl.this._protocol
+      override def transmissionProtocol: Option[Protocol] = BasicPolicyBuilder.this._protocol
 
       override def toString: String =
-        s"${name}Policy {" +
+        s"${if BasicPolicyBuilder.this.name.isDefined then BasicPolicyBuilder.this.name.get else ""}Policy { " +

```

Comment:
Like in the Policy.name value case, is it possible to use `getOrElse` to simplify this line?